### PR TITLE
[wip] Backoffice UI Uplift

### PR DESCRIPTION
**Pull request by dacook**
#### What? Why?

This is a shared development branch. We will likely rebase/rewrite history before submitting a PR to master.

- #10679

Log of changes:

1. DC: Add products_v3 page with simple table and styling
2. JB: Add async loading of products table with spinner

#### What should we test?

- Visit ... page.
- 

#### Release notes

Changelog Category: 😎 Feature toggle

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by dacook** [comment]
Rebased on top of https://github.com/openfoodfoundation/openfoodnetwork/pull/11123

File: app/views/spree/admin/shared/_product_sub_menu.html.haml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 - content_for :sub_menu do
   %ul#sub_nav.inline-menu
-    = tab :products, match_path: '/products'
+    = tab :products
     = tab :properties
     = tab :variant_overrides, url: main_app.admin_inventory_path, match_path: '/inventory'
     = tab :import, url: main_app.admin_product_import_path, match_path: '/product_import'
     - if feature?(:new_products_page, spree_current_user)
-      = tab :new_products, url: main_app.admin_new_products_path, match_path: '/new_products'
+      = tab :new_products, url: main_app.admin_new_products_path
```
Comment (jibees) [inline app/views/spree/admin/shared/_product_sub_menu.html.haml:8]: This one should be deleted I guess ;)

Comment (dacook) [reply app/views/spree/admin/shared/_product_sub_menu.html.haml:8]: Yes I've been meaning to ask!

Comment (jibees) [reply app/views/spree/admin/shared/_product_sub_menu.html.haml:8]: --> https://github.com/openfoodfoundation/openfoodnetwork/issues/11153

**Comment by dacook** [comment]
I've reset this branch based on both the open PRs (`productsV3-searching-filtering-pagination` and `admin_v3-table-tweaks`)

**Comment by jibees** [comment]
Since https://github.com/openfoodfoundation/openfoodnetwork/pull/11007 has been merged onto master, I've rebased this one onto master (only one conflicts, and still 15 commits)

**Comment by jibees** [comment]
I've created a PR with 4 commits that tweak the dropdown filters to make them compliant to v3 version: https://github.com/openfoodfoundation/openfoodnetwork/pull/11275

**Comment by jibees** [comment]
@dacook I'd propose you two create one (or more?) PR with the remaining commits of this PR in order to close it after that. Let me know what you think about it! 🙏

**Comment by dacook** [comment]
This is really strange and had me stumped earlier. I think I've narrowed it down now, see latest commit. It seems due to an inconsistency between operating systems!! 
Admittedly, I'm on an old macOS 10.15, and am overdue for an upgrade. It seems the behaviour might be fixed in later versions, so I'm wondering what version you use?

**Comment by jibees** [comment]
Ooooch. 

Seems like this test (ie. `spec/reflexes/products_reflex_spec.rb`) on my computer as well (OS 13.2.1) but because of `eq` that should probably not use for array (ie. equals reference vs. equals value). With this diff, it's fine:

```diff
diff --git a/spec/reflexes/products_reflex_spec.rb b/spec/reflexes/products_reflex_spec.rb
index a3b0f6e6f4..1fb9e0f980 100644
--- a/spec/reflexes/products_reflex_spec.rb
+++ b/spec/reflexes/products_reflex_spec.rb
@@ -24,7 +24,7 @@ describe ProductsReflex, type: :reflex do
       it "Should sort products alphabetically by default" do
         subject.run(:fetch)
 
-        expect(subject.get(:products).to_a).to eq [
+        expect(subject.get(:products).to_a).to match_array [
           product_a,
           product_b,
           product_z,
```

**Comment by jibees** [comment]
Sorry, I don't know why i was talk about `products_spec.rb:46` but it's `products_spec.rb:34`, ie. _Should sort products alphabetically by default_ that fails on my machine, due to case issue:

Capture d’écran 2023-07-26 à 07 38 16

**Comment by dacook** [comment]
Thanks for confirming. The spec actually passes in CI, so I guess Postgres on Macs behaves differently here, which is quite strange, I'm surprised we haven't encountered this before, I guess our sorting specs don't cover this case (eg [this spec](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/models/exchange_spec.rb#L257-L260) is just uppercase).

I think I'll just update the spec to avoid comparing lower/uppercase letters, and add a note on https://github.com/openfoodfoundation/openfoodnetwork/wiki/Development-Environment-Setup%3A-OS-X . Sound reasonable?

BTW A different spec did fail in the last build (see [products_spec.rb:119](https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/5663922888/job/15346406144)), but I believe it's flaky.

**Comment by jibees** [comment]
> I think I'll just update the spec to avoid comparing lower/uppercase letters, and add a note on https://github.com/openfoodfoundation/openfoodnetwork/wiki/Development-Environment-Setup%3A-OS-X . Sound reasonable?


Yes, I guess so.

**Comment by jibees** [comment]
> BTW A different spec did fail in the last build (see [products_spec.rb:119](https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/5663922888/job/15346406144)), but I believe it's flaky.

Pass locally. 👌

**Comment by jibees** [comment]
@dacook 
Do you intend to merge this branch? I'm a bit confused, and at the same time worried about this branch that's always open. 
Thanks for your insight.

**Comment by jibees** [comment]
In order to finally close this one, I've opened two PRs: #11400, and #11401

**Pull request closed without merging**